src/components/generics/links/MaskLink.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <span>
+    <span v-if="$user.isModerator" @click="toggleMask">
```

Comment:
There is probably something wrong around here because when a moderator clicks on the "Mask" link of an archived version
![image](https://user-images.githubusercontent.com/1192331/199613199-1eca2101-71fd-400f-88e5-40f955b9f37a.png)

the latter link title remains "Mask" instead of toggling to "Unmask" as expected. See:
![image](https://user-images.githubusercontent.com/1192331/199612677-10274b7d-e139-426e-8816-a116ec70fa1c.png)

After reloading the page, the link title is OK though:
![image](https://user-images.githubusercontent.com/1192331/199612950-b44fa4c7-9449-4db8-8c76-5d14af19521c.png)

Same problem when clicking on "Unmask" then.

Reply:
@brunobesson Would you have a clue about this problem? Thanks :)

Reply:
I'll try and have a look, but a bit later, sorry.

Reply:
Thanks! No hurry :)

Reply:
I do think (not totally certain though) this is a mix of 2 problems:

1. A component isn't supposed to modify the props, it's the responsibility of it's parent, which owns them, usually as data.
2. Vue2 reactivity system is limited to some edge cases, e.g. it cannot detect some array changes (because version is an item of an array).

Anyway, I do think this is more correct:

1. MaskLink component calls the API
2. MaskLink emits an event on success
3. HisoryView listens to event and updates data accordingly (it will then be propagated to MaskLink through props)

If the reactivity still doesn't work, have a look at https://v2.vuejs.org/v2/guide/reactivity.html#For-Arrays

Reply:
(I could propose code changes and test if you like, only I fear the need to run the backend 🙂 )

Reply:
Thanks! I'll test this on my side and let you know.

Reply:
I guess I should replace 
```
this.version.masked = !this.version.masked;
```
in the ``toggleMask`` method by
```
this.$emit('version-masking', {
    'masked': !this.version.masked,
    'version': this.version
});
```
but then I have not figured out how to listen to the event in the view!?

On the other that looks a bit awkward and complicated to raise an event to the view whereas it would be more convenient and logical to deal with those changes only in the component (that looks like some internal business of the component).

Actually everything is working (eg. the "This version has been masked" shows up - or hides down - as expected when clicking the toggle link). The only things that are not updated are the "Mask" / "Unmask" labels themselves.

As it's only a moderator tool, I have then simplified the code using "Toggle masking" instead, and that's it: e2c8f0c1be2f5a274d8e889a8a83573259d3603c

Reply:
Hi @asaunier.
I'll have a look at it and test it on the demo as soon as I can catch @lbesson and have the demo bakcend updated (I have release v6.13.0, but need him to migrae the database).

I took the liberty to rebase and force-push your branch on master, along with a commit to fix linting. Hope this doesn't bother you.

Reply:
It appears that `v-translate` directive with `v-if` is the cause of the problem. It can easily be circumvented ,I didn't dig the root cause.

My remark on not modifying a prop is still valid (the reactivity system proved innocent), but I understand your point.